java-extensions/udf-extensions/src/main/java/com/starrocks/udf/UDFHelper.java

```diff
@@ -234,7 +236,8 @@ private static void getDoubleTimeResult(int numRows, Time[] boxedArr, long colum
                 nulls[i] = 1;
             } else {
                 // Note: add the timezone offset back because Time#getTime() returns the GMT timestamp
-                dataArr[i] = (boxedArr[i].getTime() + timeZone.getRawOffset()) / 1000;
+                long v = boxedArr[i].getTime();
+                dataArr[i] = (v + timeZone.getOffset(v)) / 1000;
```

Comment:
I don't get the difference between getOffset and getRawOffset

Reply:
From 1945-09-13 ~ 1981-12-31, Singapore use the time zone (07:30)
From 1982-01-01 ~ now, Signapore use the time zone (08:00)

Reply:
got that